src/tests/utils/Load.test.js

```diff
@@ -1,34 +1,39 @@
 import { unzipDependencies, loadDependenciesInStorage, loadAsFHIRDefs } from '../../utils/Load';
 import { fhirdefs } from 'fsh-sushi';
-import tarStream from 'tar-stream';
 import http from 'http';
 import 'fake-indexeddb/auto';
 
 const FHIRDefinitions = fhirdefs.FHIRDefinitions;
 
 describe('#unzipDependencies', () => {
   let getSpy = jest.SpyInstance;
-  let tarSpy = jest.SpyInstance;
   let resources = [];
+
   beforeAll(() => {
     resources = [];
-    tarSpy = jest.spyOn(tarStream, 'extract').mockImplementation(() => {
-      return undefined;
-    });
     getSpy = jest.spyOn(http, 'get').mockImplementation(() => {
       return 'hl7.fhir.r4.core-4.0.1.tgz';
     });
   });
+
   beforeEach(() => {
-    tarSpy.mockClear();
     getSpy.mockClear();
   });
+
   it('should make an http request and extract data from the resulting zip folder', () => {
     unzipDependencies(resources, 'hl7.fhir.r4.core', '4.0.1');
     const callbackFunction = getSpy.mock.calls[0][1];
     expect(getSpy).toBeCalled();
     expect(getSpy).toBeCalledWith('https://packages.fhir.org/hl7.fhir.r4.core/4.0.1', callbackFunction);
   });
+
+  it('should add failed http requests to a list of emptyDependencies', async () => {
+    const unzipPromise = unzipDependencies(resources, 'hello', '123');
+    const callbackFunction = getSpy.mock.calls[0][1];
+    callbackFunction({ statusCode: 404 });
+    expect(getSpy).toBeCalled();
+    expect(unzipPromise).resolves.toEqual({ emptyDependencies: ['hello123'], resourceArr: resources });
```

Comment:
I think this last expect statement might need to be `await`ed. I'm not 100% I understand why, except that I tried commenting out the line that pushes to the `emptyDependencies` array and the test still passed, although it did show that the two objects didn't match. Awaiting the last expect does have the test pass and fail as expected.